api/response/schema.py

```diff
@@ -251,6 +279,46 @@ def mutate(self, info, responses, resolved):
         )
 
 
+class ArchiveResponse(graphene.Mutation):
+    """
+    Returns payload after marking a resolved response
+    as archived
+    """
+
+    class Arguments:
+        response_id = graphene.Int(required=True)
+
+    room_response = graphene.Field(ResponseDetail)
+
+    @Auth.user_roles('Admin')
+    def mutate(self, info, response_id):
+        responses_query = Response.get_query(info)
+        resolved_responses = responses_query.filter(
+            ResponseModel.resolved)
+        active_responses = resolved_responses.filter(
+            ResponseModel.state == "active")
+        exact_response = active_responses.filter(
+            ResponseModel.id == response_id).first()
```

Comment:
You can combine `296` to `301` by passing the filters for `exact_response` in `296`

Reply:
Changes made as requested.